src/main/java/Config/Routes/RouteFactory.java

```diff
@@ -8,12 +8,16 @@
 public class RouteFactory {
 
 	public Route createRoute(Method method, String path, Handler handler) {
-		Route route = null;
-		if (method == GET) {
-			route = new GETRoute(method, path, handler);
-		} else if (method == POST) {
-			route = new POSTRoute(method, path, handler);
-		}
-		return route;
-	}
+      Route route = null;
+      switch (method) {
+        case GET:
+          return new GETRoute(method, path, handler);
+        case POST:
+          return new POSTRoute(method, path, handler);
+        case OPTIONS:
+          return new OPTIONSRoute(method, path, handler);
+        default:
+          throw new Error();
+      }
```

Comment:
Re: my above feedback, if you go with a generic structure, here you could reduce the switch statement to this:

```java
return new Route(method, path, handler);
```

And just trust your `Method.toMethod` approach with the error handling.